src/DotVVM.Framework/Resources/Scripts/DotVVM.ts

```diff
@@ -872,23 +872,57 @@ class DotVVM {
         ko.bindingHandlers["dotvvm-UpdateProgress-Visible"] = {
             init(element: any, valueAccessor: () => any, allBindingsAccessor: KnockoutAllBindingsAccessor, viewModel: any, bindingContext: KnockoutBindingContext) {
                 element.style.display = "none";
+                var delay = element.getAttribute("data-delay");
+                var timeout;
+                var running = false;
+
+                var show = () => {
+                    running = true;
+                    if (delay == null) {
+                        element.style.display = "";
+                    } else {
+                        timeout = setTimeout(e => {
+                            element.style.display = "";
+                        }, delay);
```

Comment:
By the way, the property could be bindable, if you'd read the "data-delay" attribute on each run and render an `attr` binding in case there is a binding (which can be simply done by assigning the `GetValueRaw(DelayProperty)` to `this.Attributes["data-delay"]`). But I don't see any use case for this...